### DISCUSSION TITLE
Help features

### DISCUSSION THREAD

**Discussion by zakblacki** [AntlerVC/firetable#713] [Support / Q&A] [answered]
Hi, need some guidance :
- We cannot see the document Id (row)  (if user want to create a new row we should be allowed to set the name or id manually not auto generate)
- There is no search feature available (to find some specific data in our collection)
- Also we need feature to upload images to storage and get link from it 
- It's annoying to have to edit Field type (give option for users if they want to change auto or not Urgent)

**Comment by harinij** [comment, marked as answer]
Hi @zakblacki 👋

- You have the option to make row document IDs be either automatically generated (small id/random) or manually. You can set this on your table by clicking the arrow right next to the "Add row" button. By default it automatically sets it to a small id that makes all new rows show up to the top. 
<img width="579" alt="Screen Shot 2022-07-19 at 4 45 51 pm" src="https://user-images.githubusercontent.com/307298/179683987-23931d5d-5a3f-44fe-9ac5-b29ed05a22a8.png">

- We will be adding a full text search feature soon, in the meantime only filters options is available.
- For uploading images or files to storage, you can use the "Image" or "Files" column and simply add your images/files there. This will automatically upload the image to storage, you can get the URL if you right-click on the image. If you want the URL in its own column, then you can create a derivative column to return the value of the URL `row.productImage[0].downloadURL`. You can see this example in this ImageURL column of this [demo app](https://demo.rowy.io/table/boardGames).
- Can you share a bit more about in which context you are changing the field types? Is it during bulk import or while adding a new table/columns?

**Reply by zakblacki** [reply]
Thanks for replying, 
- Add row document ID is not working for me it's grayed out 
- As for uploading images the field turn to be an array nested (I only needed a string ) but still works for now
- And Yes the field types changes during bulk import 

![image](https://user-images.githubusercontent.com/29728429/179689914-d1e9cafe-f5ef-44ac-8309-25bf57534ee8.png)

**Reply by notsidney** [reply]
Hi, this is disabled here because this is a collection group table. You can see further discussion here: #289

**Comment by zakblacki** [comment]
Hi, once again I require assistance 
How do we add new row line because it's not generating anything when I click on these 3 options 

![image](https://user-images.githubusercontent.com/29728429/187047099-d830ee84-bf72-41cc-b5fb-37ab045cb78f.png)

**Reply by notsidney** [reply]
Hi, those three options change what the main “Add row” button does. Click “Add row” again after choosing an option.

**Reply by notsidney** [reply]
Additionally, please create a new discussion item instead of reusing this one.
